Scripts DICOM/writeTROTSToDICOM.py

```diff
@@ -215,6 +224,267 @@
+                    icpoi.TableTopPitchAngle = 0
+                    icpoi.TableTopPitchRotationDirection = 'NONE'
+                    icpoi.TableTopRollAngle = 0
+                    icpoi.TableTopRollRotationDirection = 'NONE'
+                    icpoi.GantryPitchAngle = 0
+                    icpoi.GantryPitchRotationDirection = 'NONE'
+                    icpoi.SnoutPosition = None
+                    icpoi.ScanSpotTuneID = 'TROTS_1.0'
+                    icpoi.NumberOfScanSpotPositions = len(controlpointinfo["MetersetWeights"])
+                    icpoi.ScanSpotPositionMap = controlpointinfo["ScanSpotPositions"]
+                    icpoi.ScanSpotMetersetWeights = controlpointinfo["MetersetWeights"]
+                    sigma1 = np.interp(controlpointinfo["BeamEnergy"],beamSigmaEnergy[:][0], beamSigmas[:][0])
+                    sigma2 = np.interp(controlpointinfo["BeamEnergy"],beamSigmaEnergy[:][1], beamSigmas[:][1])
+                    icpoi.ScanningSpotSize = [sigma1,sigma2]
+                    icpoi.NumberOfPaintings = 1
+                    if((beaminfo["ConstantRangeShifter"]==False) or ((controlpointinfo["ControlPointNumber"]==0) and (controlpointinfo["RangeShifter"]!=0))):
+                        icpoi.RangeShifterSettingsSequence = Sequence()
+                        rsSettings = Dataset()
+                        if(controlpointinfo["RangeShifter"]==0):
+                            rsSettings.RangeShifterSetting = "OUT"
+                            rsSettings.RangeShifterWaterEquivalentThickness = beaminfo["RangeShifters"][0]
+                            rsSettings.ReferencedRangeShifterNumber = 0
+                        else:
+                            rsSettings.RangeShifterSetting = "IN"
+                            rsSettings.RangeShifterWaterEquivalentThickness = controlpointinfo["RangeShifter"]
+                            rsSettings.ReferencedRangeShifterNumber = beaminfo["RangeShifters"].index(controlpointinfo["RangeShifter"])
+                        icpoi.RangeShifterSettingsSequence.append(rsSettings)
+                    assert(abs(icpoi.CumulativeMetersetWeight - totalMetersetWeightOfControlPoints) < 1e-9)
+                    if(icpoi.NumberOfScanSpotPositions != 1):
+                        totalMetersetWeightOfControlPoints += sum(icpoi.ScanSpotMetersetWeights)
+                    else:
+                        totalMetersetWeightOfControlPoints += icpoi.ScanSpotMetersetWeights
+                    be.IonControlPointSequence.append(icpoi)
+                    
+                    icpoi = Dataset()
+                    icpoi.NominalBeamEnergyUnit = 'MEV'
+                    icpoi.ControlPointIndex = 2*controlpointinfo["ControlPointNumber"] + 1
+                    icpoi.NominalBeamEnergy = controlpointinfo["BeamEnergy"]
+                    icpoi.CumulativeMetersetWeight = controlpointinfo["CumulativeMetersetWeight"]+ sum(controlpointinfo["MetersetWeights"])

```

Comment:
```suggestion
                    icpoi.NominalBeamEnergy = format_number_as_ds(controlpointinfo["BeamEnergy"])
                    icpoi.CumulativeMetersetWeight = format_number_as_ds(controlpointinfo["CumulativeMetersetWeight"]+ sum(controlpointinfo["MetersetWeights"]))
```

Rebase your branch and rerun the script, you will then see that you need this function to not violate the DICOM standard.

Reply:
Ok so do i need to do this everywhere that I have floats?

Reply:
It still doesn't seem to throw any errors so didn't know I needed this